pkg/pillar/types/assignableadapters.go

```diff
@@ -533,6 +659,7 @@ func (aa *AssignableAdapters) CheckBadUSBBundles() {
 			continue
 		}
 
+		ioBundle.Error.removeByType(ioBundleCollisionErr{})
```

Comment:
My understanding (which could be wrong) is that when there is a collision the collision error is added to both of the colliding bundles. Is that correct?

If it is, then wouldn't you want to remove all the collisions before scanning for new collisions i.e., do the removal in a separate loop?

Reply:
> the collision error is added to both of the colliding bundles

That is correct.


> do the removal in a separate loop?

I think it is not a problem doing it this way, because the `cycleDetectedError` is added in a separate loop.
But I will change it, because indeed it makes it more readable.